src/service/dependencies.py

```diff
@@ -0,0 +1,27 @@
+"""
+Dependencies for FastAPI dependency injection.
+"""
+
+import os
+
+from pyspark.sql import SparkSession
+
+from src.delta_lake.spark_utils import get_spark_session as _get_spark_session
+from src.service.http_bearer import KBaseHTTPBearer
+
+# Initialize the KBase auth dependency for use in routes
+auth = KBaseHTTPBearer()
+
+DEFAULT_SPARK_POOL = "default"
+DEFAULT_EXECUTOR_CORES = 1
+
+
+def get_spark_session() -> SparkSession:
```

Comment:
Out of curiosity why are you using the dependency system for this? I only used it for auth because that's how you get the auth stuff to show up in the generated UI. Otherwise I probably would've just made it a method call

Reply:
no specific reason - it works just fine.